### PR TITLE
chore: add custom image metadata option for iOS and add missing types for custom metadata properties

### PR DESCRIPTION
**Pull request by Duell10111**
#### Update the documentation
Updated the API.md with the new imageMetadataUri

#### Provide an example of how to test the change
Can be tested in the tvOS example of the app

#### Describe the changes
Added a possibillity to modify the image metadata of a video shown in tvOS for example or the Now Playing area in iOS.

Created as descripted in the documentation of apple:
https://developer.apple.com/documentation/avkit/customizing_the_tvos_playback_experience#3820220

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -981,7 +981,7 @@ Platforms: iOS, Android
 
 ##### Overriding the metadata of a source
 
-Provide an optional `title`, `subtitle`  and/or `description` properties for the video. 
+Provide an optional `title`, `subtitle`, `customImageUri` and/or `description` properties for the video.
```
Comment (freeboub) [inline API.md:984]: Is there any limitation for the image ?
I see the internal tvOS api need a png, but not clear if a png is mandatory in the API

**Review by freeboub — COMMENTED** [review]
one minor clarification expected, else this is ok for me

**Comment by freeboub** [comment]
notice also that you have linter error ...
Run yarn lint
yarn run v1.22.19
$ yarn eslint .
$ /home/runner/work/react-native-video/react-native-video/node_modules/.bin/eslint .

/home/runner/work/react-native-video/react-native-video/src/Video.tsx
Error:   [4](https://github.com/react-native-video/react-native-video/actions/runs/6440734579/job/17489846310?pr=3280#step:5:5)8[5](https://github.com/react-native-video/react-native-video/actions/runs/6440734579/job/17489846310?pr=3280#step:5:6):11  error  Do not use "@ts-ignore" because it alters compilation errors  @typescript-eslint/ban-ts-comment

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.

You can keep the sample as is. @KrzysztofMoch said you will have a look to the sample !
Thank you

File: API.md (hunk 1 above)
Comment (Duell10111) [reply API.md:984]: No the API supports the image formats supported by UIImage, which contains the classic ones like jpeg, png.
The list can be found here:

https://developer.apple.com/library/archive/documentation/2DDrawing/Conceptual/DrawingPrintingiOS/LoadingImages/LoadingImages.html

**Comment by Duell10111** [comment]
> notice also that you have linter error ...
> 
> Run yarn lint
> 
> yarn run v1.22.19
> 
> $ yarn eslint .
> 
> $ /home/runner/work/react-native-video/react-native-video/node_modules/.bin/eslint .
> 
> 
> 
> /home/runner/work/react-native-video/react-native-video/src/Video.tsx
> 
> Error:   [4](https://github.com/react-native-video/react-native-video/actions/runs/6440734579/job/17489846310?pr=3280#step:5:5)8[5](https://github.com/react-native-video/react-native-video/actions/runs/6440734579/job/17489846310?pr=3280#step:5:6):11  error  Do not use "@ts-ignore" because it alters compilation errors  @typescript-eslint/ban-ts-comment
> 
> 
> 
> ✖ 1 problem (1 error, 0 warnings)
> 
> 
> 
> error Command failed with exit code 1.
> 
> info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> 
> error Command failed with exit code 1.
> 
> info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> 
> Error: Process completed with exit code 1.
> 
> 
> 
> You can keep the sample as is. @KrzysztofMoch said you will have a look to the sample !
> 
> Thank you

I only added this line as the selectedtexttracks contained a type issue, blocking the typescript compile operation.
I can remove this line, once I am home again. 😊

**Comment by KrzysztofMoch** [comment]
@Duell10111 There is fix on master for this ts error and for eslint errors, you can make rebase 😁

File: src/Video.tsx [hunk 2]
```diff
@@ -478,6 +482,7 @@ const Video = forwardRef<VideoRef, ReactVideoProps>(
             _restoreUserInterfaceForPIPStopCompletionHandler
           }
           textTracks={textTracks}
+          // @ts-ignore
```
Comment (freeboub) [inline src/Video.tsx]:
```suggestion
```

**Comment by freeboub** [comment]
I was able to fix it :) So let's merge !
(And thank you for image detailed informations !)
Thank you !

**Pull request merged**